searchlib/src/vespa/searchlib/engine/proto_rpc_adapter.cpp

```diff
@@ -0,0 +1,274 @@
+struct PingCompletionHandler : MonitorClient {
+    FRT_RPCRequest &req;
+    PingCompletionHandler(FRT_RPCRequest &req_in) : req(req_in) {}
+    void pingDone(MonitorReply::UP reply) override {
+        ProtoMonitorReply msg;
+        ProtoConverter::monitor_reply_to_proto(*reply, msg);
+        encode_message(msg, *req.GetReturn());
+        req.Return();
+    }
+};
+
+//-----------------------------------------------------------------------------
+
+}
+
+ProtoRpcAdapter::ProtoRpcAdapter(SearchServer &search_server,
+                                 DocsumServer &docsum_server,
+                                 MonitorServer &monitor_server,
+                                 FRT_Supervisor &orb)
+    : _search_server(search_server),
+      _docsum_server(docsum_server),
+      _monitor_server(monitor_server)
+{
+    FRT_ReflectionBuilder rb(&orb);
+    //-------------------------------------------------------------------------
+    rb.DefineMethod("vespa.searchprotocol.search", "bix", "bix",
+                    FRT_METHOD(ProtoRpcAdapter::rpc_search), this);
+    rb.MethodDesc("perform a search against this back-end");
+    rb.ParamDesc("encoding", "0=raw, 6=lz4, 7=zstd");
+    rb.ParamDesc("uncompressed_size", "uncompressed size of serialized request");
+    rb.ParamDesc("request", "possibly compressed serialized request");
+    rb.ReturnDesc("encoding",  "0=raw, 6=lz4, 7=zstd");
+    rb.ReturnDesc("uncompressed_size", "uncompressed size of serialized reply");
+    rb.ReturnDesc("reply", "possibly compressed serialized reply");
+    //-------------------------------------------------------------------------
+    rb.DefineMethod("vespa.searchprotocol.getDocsums", "bix", "bix",
+                    FRT_METHOD(ProtoRpcAdapter::rpc_getDocsums), this);
+    rb.MethodDesc("fetch document summaries from this back-end");
+    rb.ParamDesc("encoding", "0=raw, 6=lz4, 7=zstd");
+    rb.ParamDesc("uncompressed_size", "uncompressed size of serialized request");
+    rb.ParamDesc("request", "possibly compressed serialized request");
+    rb.ReturnDesc("encoding",  "0=raw, 6=lz4, 7=zstd");
+    rb.ReturnDesc("uncompressed_size", "uncompressed size of serialized reply");
+    rb.ReturnDesc("reply", "possibly compressed serialized reply");
+    //-------------------------------------------------------------------------
+    rb.DefineMethod("vespa.searchprotocol.ping", "bix", "bix",
+                    FRT_METHOD(ProtoRpcAdapter::rpc_ping), this);
+    rb.MethodDesc("perform a search against this back-end");
+    rb.ParamDesc("encoding", "0=raw, 6=lz4, 7=zstd");
+    rb.ParamDesc("uncompressed_size", "uncompressed size of serialized request");
+    rb.ParamDesc("request", "possibly compressed serialized request");
+    rb.ReturnDesc("encoding",  "0=raw, 6=lz4, 7=zstd");
+    rb.ReturnDesc("uncompressed_size", "uncompressed size of serialized reply");
+    rb.ReturnDesc("reply", "possibly compressed serialized reply");
```

Comment:
Since the param/return signatures and descriptions are the same for all RPCs accepting and returning protobuf stuff, could consider factoring some of it out into a separate function.

Reply:
fixed as part of force push (in order to make travis compile with new and better protobuf integration)